ntoskrnl/mm/ARM3/expool.c

```diff
@@ -1704,8 +1739,20 @@ ExpFindAndRemoveTagBigPages(IN PVOID Va,
     // the lock and return the tag that was located
     //
     InterlockedIncrement((PLONG)&Entry->Va);
-    InterlockedDecrementUL(&ExpPoolBigEntriesInUse);
-    KeReleaseSpinLock(&ExpLargePoolTableLock, OldIrql);
+
+    ExpPoolBigEntriesInUse--;
+
+    /* If reaching 12.5% of the size (or whatever integer rounding gets us to),
+     * halve the allocation size, which will get us to 25% of space used. */
```

Comment:
Are these numbers connected with 75% from above?
I think it makes sense to create a define so the value can be easily tuned. Somebody may want to do a benchmark for that and find the best value :D

Reply:
Well, this is just what happens when you play with halving & doubling sizes.

The idea is to keep the occupation rate in the 25% - 75% range. This could as well be 33 - 66, 12 - 88 or whatever.

I'll see to have some defines.

Reply:
Done